# Analysis on COVID-19 Vaccinations and Deaths in the United States



In this analysis, I will be comparing COVID-19 vaccination rates and COVID-19 deaths and see if they have any correlation. I will be using two data sets in this analysis: 
- Our World In Data: https://ourworldindata.org/us-states-vaccinations
    - State-by-state data on COVID-19 vaccinations in the United States
- CDC: https://data.cdc.gov/Case-Surveillance/United-States-COVID-19-Cases-and-Deaths-by-State-o/9mfq-cb36
    - State-by-state data on COVID-19 excess deaths associated with COVID-19 in the United States

In [193]:
#all necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly 
import plotly.express as px
import plotly.graph_objects as go
import plotly.offline as py
from plotly.offline import iplot
from plotly.subplots import make_subplots
import plotly.figure_factory as ff

In [194]:
#read in data set

#1) Our World In Data Data set
df1 = pd.read_csv("us_state_vaccinations.csv")

#2) CDC Data Set
df2 = pd.read_csv("United_States_COVID-19_Cases_and_Deaths_by_State_over_Time.csv")

## Exploring the Datasets

In [195]:
df1.head()

date location  total_vaccinations  total_distributed  \
0  2021-01-12  Alabama             78134.0           377025.0   
1  2021-01-13  Alabama             84040.0           378975.0   
2  2021-01-14  Alabama             92300.0           435350.0   
3  2021-01-15  Alabama            100567.0           444650.0   
4  2021-01-16  Alabama                 NaN                NaN   

   people_vaccinated  people_fully_vaccinated_per_hundred  \
0            70861.0                                 0.15   
1            74792.0                                 0.19   
2            80480.0                                  NaN   
3            86956.0                                 0.28   
4                NaN                                  NaN   

   total_vaccinations_per_hundred  people_fully_vaccinated  \
0                            1.59                   7270.0   
1                            1.71                   9245.0   
2                            1.88                      NaN   
3                            2.05                  13488.0   
4                             NaN                      NaN   

   people_vaccinated_per_hundred  distributed_per_hundred  \
0                           1.45                     7.69   
1                           1.53                     7.73   
2                           1.64                     8.88   
3                           1.77                     9.07   
4                            NaN                      NaN   

   daily_vaccinations_raw  daily_vaccinations  daily_vaccinations_per_million  \
0                     NaN                 NaN                             NaN   
1                  5906.0              5906.0                          1205.0   
2                  8260.0              7083.0                          1445.0   
3                  8267.0              7478.0                          1525.0   
4                     NaN              7498.0                          1529.0   

   share_doses_used  total_boosters  total_boosters_per_hundred  
0             0.207             NaN                         NaN  
1             0.222             NaN                         NaN  
2             0.212             NaN                         NaN  
3             0.226             NaN                         NaN  
4               NaN             NaN                         NaN

In [196]:
df2.head()

submission_date state  tot_cases  conf_cases  prob_cases  new_case  \
0      12/01/2021    ND     163565    135705.0     27860.0       589   
1      08/17/2020    MD     100715         NaN         NaN       503   
2      03/28/2022    VT     107785         NaN         NaN       467   
3      03/18/2020    ME         44        44.0         0.0        12   
4      08/29/2021    WA     556639         NaN         NaN      2991   

   pnew_case  tot_death  conf_death  prob_death  new_death  pnew_death  \
0      220.0       1907         NaN         NaN          9         0.0   
1        0.0       3765      3616.0       149.0          3         0.0   
2       35.0        585         NaN         NaN          0         0.0   
3        0.0          0         0.0         0.0          0         0.0   
4      479.0       6525         NaN         NaN          9         0.0   

               created_at consent_cases consent_deaths  
0  12/02/2021 02:35:20 PM         Agree      Not agree  
1  08/19/2020 12:00:00 AM           NaN          Agree  
2  03/29/2022 01:30:11 PM     Not agree      Not agree  
3  03/20/2020 12:00:00 AM         Agree          Agree  
4  08/31/2021 12:00:00 AM           NaN            NaN

In [197]:
df1.tail()

date location  total_vaccinations  total_distributed  \
29151  2022-04-03  Wyoming            740229.0           949085.0   
29152  2022-04-04  Wyoming            740234.0           949085.0   
29153  2022-04-05  Wyoming            741946.0           949785.0   
29154  2022-04-06  Wyoming            742407.0           950585.0   
29155  2022-04-07  Wyoming            743110.0           952485.0   

       people_vaccinated  people_fully_vaccinated_per_hundred  \
29151           338155.0                                51.10   
29152           338162.0                                51.10   
29153           338316.0                                51.20   
29154           338372.0                                51.21   
29155           338512.0                                51.23   

       total_vaccinations_per_hundred  people_fully_vaccinated  \
29151                          127.90                 295755.0   
29152                          127.90                 295760.0   
29153                          128.20                 296330.0   
29154                          128.28                 296382.0   
29155                          128.40                 296503.0   

       people_vaccinated_per_hundred  distributed_per_hundred  \
29151                          58.43                   163.99   
29152                          58.43                   163.99   
29153                          58.46                   164.11   
29154                          58.47                   164.25   
29155                          58.49                   164.57   

       daily_vaccinations_raw  daily_vaccinations  \
29151                    54.0               325.0   
29152                     5.0               324.0   
29153                  1712.0               523.0   
29154                   461.0               507.0   
29155                   703.0               565.0   

       daily_vaccinations_per_million  share_doses_used  total_boosters  \
29151                           562.0             0.780        126103.0   
29152                           560.0             0.780        126105.0   
29153                           904.0             0.781        126347.0   
29154                           876.0             0.781        126422.0   
29155                           976.0             0.780        126581.0   

       total_boosters_per_hundred  
29151                       21.79  
29152                       21.79  
29153                       21.83  
29154                       21.84  
29155                       21.87

In [198]:
df2.tail()

submission_date state  tot_cases  conf_cases  prob_cases  new_case  \
48355      12/15/2020    DC      25339         NaN         NaN       301   
48356      01/12/2022    WY     123743     97745.0     25998.0       989   
48357      07/11/2020    RI      16456         NaN         NaN        52   
48358      11/22/2021    AZ    1245127   1127692.0    117435.0      3249   
48359      04/16/2020    WY        401       296.0       105.0         8   

       pnew_case  tot_death  conf_death  prob_death  new_death  pnew_death  \
48355        0.0        720         NaN         NaN          4         0.0   
48356      246.0       1588      1588.0         0.0          0         0.0   
48357        0.0        978         NaN         NaN          3         0.0   
48358      403.0      21942     19414.0      2528.0          2         0.0   
48359        0.0          2         2.0         0.0          0         0.0   

                   created_at consent_cases consent_deaths  
48355  12/16/2020 02:28:57 PM           NaN            NaN  
48356  01/13/2022 02:34:51 PM         Agree          Agree  
48357  03/25/2022 01:13:04 PM     Not agree            NaN  
48358  11/23/2021 02:18:53 PM         Agree          Agree  
48359  04/16/2020 04:22:39 PM         Agree          Agree

In [199]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29156 entries, 0 to 29155
Data columns (total 16 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   date                                 29156 non-null  object 
 1   location                             29156 non-null  object 
 2   total_vaccinations                   26404 non-null  float64
 3   total_distributed                    26144 non-null  float64
 4   people_vaccinated                    26140 non-null  float64
 5   people_fully_vaccinated_per_hundred  24782 non-null  float64
 6   total_vaccinations_per_hundred       24881 non-null  float64
 7   people_fully_vaccinated              26060 non-null  float64
 8   people_vaccinated_per_hundred        24857 non-null  float64
 9   distributed_per_hundred              24861 non-null  float64
 10  daily_vaccinations_raw               25378 non-null  float64
 11  daily_vaccinations          

In [200]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48360 entries, 0 to 48359
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   submission_date  48360 non-null  object 
 1   state            48360 non-null  object 
 2   tot_cases        48360 non-null  int64  
 3   conf_cases       26643 non-null  float64
 4   prob_cases       26571 non-null  float64
 5   new_case         48360 non-null  int64  
 6   pnew_case        44658 non-null  float64
 7   tot_death        48360 non-null  int64  
 8   conf_death       26272 non-null  float64
 9   prob_death       26272 non-null  float64
 10  new_death        48360 non-null  int64  
 11  pnew_death       44690 non-null  float64
 12  created_at       48360 non-null  object 
 13  consent_cases    40295 non-null  object 
 14  consent_deaths   41106 non-null  object 
dtypes: float64(6), int64(4), object(5)
memory usage: 5.5+ MB


In [201]:
df1.describe()

total_vaccinations  total_distributed  people_vaccinated  \
count        2.640400e+04       2.614400e+04       2.614000e+04   
mean         1.041567e+07       1.286061e+07       5.476839e+06   
std          4.539669e+07       5.595092e+07       2.329017e+07   
min          4.160000e+02       6.000000e+03       4.010000e+02   
25%          7.980258e+05       1.025415e+06       4.625658e+05   
50%          2.796562e+06       3.450545e+06       1.543121e+06   
75%          7.156782e+06       8.510450e+06       3.631310e+06   
max          5.639991e+08       7.084503e+08       2.559757e+08   

       people_fully_vaccinated_per_hundred  total_vaccinations_per_hundred  \
count                         24782.000000                    24881.000000   
mean                             42.865106                       99.337657   
std                              21.569616                       50.880264   
min                               0.000000                        0.170000   
25%                              28.730000                       64.490000   
50%                              47.355000                      102.080000   
75%                              58.520000                      135.600000   
max                              87.760000                      222.590000   

       people_fully_vaccinated  people_vaccinated_per_hundred  \
count             2.606000e+04                   24857.000000   
mean              4.486172e+06                      51.700431   
std               1.955768e+07                      22.735779   
min               1.000000e+00                       0.240000   
25%               3.031015e+05                      37.590000   
50%               1.159176e+06                      55.210000   
75%               3.050392e+06                      67.350000   
max               2.181356e+08                     100.600000   

       distributed_per_hundred  daily_vaccinations_raw  daily_vaccinations  \
count             24861.000000            2.537800e+04        2.909100e+04   
mean                122.831304            4.045971e+04        3.921653e+04   
std                  59.643171            1.932883e+05        1.795258e+05   
min                   6.140000            0.000000e+00        0.000000e+00   
25%                  84.900000            1.326250e+03        2.363000e+03   
50%                 124.600000            7.859000e+03        9.327000e+03   
75%                 167.510000            2.465425e+04        2.441200e+04   
max                 280.460000            4.629928e+06        3.384387e+06   

       daily_vaccinations_per_million  share_doses_used  total_boosters  \
count                    27438.000000      26144.000000    9.238000e+03   
mean                      3653.621838          0.792863    2.027009e+06   
std                       2738.608534          0.117025    8.777949e+06   
min                          0.000000          0.043000    8.000000e+00   
25%                       1738.000000          0.755000    1.646490e+05   
50%                       2890.500000          0.804000    5.621105e+05   
75%                       4870.000000          0.856000    1.448513e+06   
max                      27652.000000          1.138000    9.842474e+07   

       total_boosters_per_hundred  
count                 8788.000000  
mean                    19.916246  
std                     10.970407  
min                      0.010000  
25%                     10.850000  
50%                     20.165000  
75%                     27.590000  
max                     52.450000

In [202]:
df2.describe()

tot_cases    conf_cases     prob_cases       new_case  \
count  4.836000e+04  2.664300e+04   26571.000000   48360.000000   
mean   4.575271e+05  4.717415e+05   68681.618456    1646.044913   
std    8.392395e+05  7.764371e+05  107786.375493    5296.282952   
min    0.000000e+00  0.000000e+00       0.000000  -10199.000000   
25%    8.704750e+03  4.506200e+04      46.000000      27.000000   
50%    1.410770e+05  2.084890e+05   14096.000000     380.000000   
75%    5.662822e+05  6.398790e+05  100822.500000    1424.000000   
max    9.105181e+06  8.503930e+06  602704.000000  319809.000000   

           pnew_case     tot_death    conf_death    prob_death     new_death  \
count   44658.000000  48360.000000  26272.000000  26272.000000  48360.000000   
mean      263.225133   7487.334615   7727.311320    856.407240     19.952895   
std      1559.424472  12194.798981   9693.196417   1279.032256     47.268477   
min   -171804.000000      0.000000      0.000000      0.000000   -352.000000   
25%         0.000000    184.000000    889.000000      0.000000      0.000000   
50%         4.000000   2330.000000   4008.000000    236.000000      4.000000   
75%       171.000000   9452.000000  11150.500000   1212.000000     20.000000   
max    171617.000000  88355.000000  71408.000000   6725.000000   1178.000000   

         pnew_death  
count  44690.000000  
mean       2.043656  
std       27.430908  
min    -2594.000000  
25%        0.000000  
50%        0.000000  
75%        1.000000  
max     2919.000000

In [203]:
df1.location.unique()

array(['Alabama', 'Alaska', 'American Samoa', 'Arizona', 'Arkansas',
       'Bureau of Prisons', 'California', 'Colorado', 'Connecticut',
       'Delaware', 'Dept of Defense', 'District of Columbia',
       'Federated States of Micronesia', 'Florida', 'Georgia', 'Guam',
       'Hawaii', 'Idaho', 'Illinois', 'Indian Health Svc', 'Indiana',
       'Iowa', 'Kansas', 'Kentucky', 'Long Term Care', 'Louisiana',
       'Maine', 'Marshall Islands', 'Maryland', 'Massachusetts',
       'Michigan', 'Minnesota', 'Mississippi', 'Missouri', 'Montana',
       'Nebraska', 'Nevada', 'New Hampshire', 'New Jersey', 'New Mexico',
       'New York State', 'North Carolina', 'North Dakota',
       'Northern Mariana Islands', 'Ohio', 'Oklahoma', 'Oregon',
       'Pennsylvania', 'Puerto Rico', 'Republic of Palau', 'Rhode Island',
       'South Carolina', 'South Dakota', 'Tennessee', 'Texas',
       'United States', 'Utah', 'Vermont', 'Veterans Health',
       'Virgin Islands', 'Virginia', 'Washington', 'West V

In [204]:
df2.state.unique()

array(['ND', 'MD', 'VT', 'ME', 'WA', 'DE', 'WI', 'GU', 'MS', 'AL', 'NV',
       'MI', 'CT', 'NC', 'IN', 'NE', 'MO', 'NH', 'ID', 'IL', 'MT', 'CA',
       'VI', 'OR', 'FSM', 'NJ', 'DC', 'MN', 'AZ', 'RI', 'LA', 'KY', 'SC',
       'VA', 'WY', 'KS', 'FL', 'CO', 'WV', 'AR', 'MP', 'AS', 'HI', 'AK',
       'GA', 'OK', 'PW', 'TX', 'PR', 'UT', 'MA', 'NYC', 'RMI', 'NY', 'SD',
       'OH', 'PA', 'NM', 'TN', 'IA'], dtype=object)

We can see that there have been 563,999,093 total doses given and 255.975,678 total people fully vaccinated in the United States.

In [205]:
df1[df1["total_vaccinations"] == df1["total_vaccinations"].max()][["date", "location","total_vaccinations"]]

date       location  total_vaccinations
25096  2022-04-07  United States         563999093.0

In [206]:
df1[df1["people_vaccinated"] == df1["people_vaccinated"].max()][["date", "location","people_vaccinated"]]

date       location  people_vaccinated
25096  2022-04-07  United States        255975678.0

We can see that the highest death total is in California, with 88,355 deaths. 

In [211]:
df2[df2["tot_death"] == df2["tot_death"].max()][["submission_date", "state","tot_death"]]

submission_date state  tot_death
281        04/06/2022    CA      88355
35372      04/05/2022    CA      88355

In [207]:
fig = px.histogram(data_frame=df1, x=df1["date"],y=df1["total_vaccinations"], labels={"date":"Date","total_vaccinations":"Total Vaccinations"}, title="Total Vaccination Doses Over Time")
fig.show()

0        12/01/2021
1        08/17/2020
2        03/28/2022
3        03/18/2020
4        08/29/2021
            ...    
48355    12/15/2020
48356    01/12/2022
48357    07/11/2020
48358    11/22/2021
48359    04/16/2020
Name: submission_date, Length: 48360, dtype: object